File: kangaroo_app_sdk_android/android/src/main/kotlin/com/kangaroo/flutterplugin/android/KangarooAppSdk.kt
package com.kangaroo.flutterplugin.android

import android.content.Context
import com.kangaroo.flutterplugin.android.base.PluginChannelHandler
import com.kangaroo.flutterplugin.android.base.pushSerializedResultToEventSink
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import kangaroorewards.appsdk.core.KangarooSdk

class KangarooAppSdk {
    var sink: EventChannel.EventSink? = null

    companion object {
        fun initializeSdk(call: MethodCall, context: Context): Unit? {
            KangarooSdk(context).initialize(
                appId = call.argument<String?>("appId") ?: "",
                environment = call.argument<String?>("environment") ?: "",
                firebaseAppCheckToken = call.argument<String?>("firebaseAppCheckToken") ?: "",
                firebaseAuthToken = call.argument<String?>("firebaseAuthToken") ?: "",
            )
            return null
        }

        fun getSession(call: MethodCall, context: Context): String? {
            return KangarooSdk(context).getSession()
        }

        fun killSession(call: MethodCall, context: Context): Unit? {
            KangarooSdk(context).killSession()
            return null
        }

        fun setPreferredLanguage(call: MethodCall, context: Context): Unit? {
            val language  = call.argument<String?>("preferredLanguage") ?: ""
            println("preferred language android: $language")
            KangarooSdk(context).setPreferredLanguage(language)
            return null;
        }

        fun getPreferredLanguage(call: MethodCall, context: Context): String? {
            return KangarooSdk(context).getPreferredLanguage()
        }

    }
}

File: kangaroo_app_sdk/lib/kangaroo_app_sdk.dart
import 'package:kangaroo_app_sdk_platform_interface/platform_interface/kangaroo_app_sdk_platform_interface.dart';

class KangarooAppSdk {
  static initializeSdk(
    String appId,
    String? firebaseAppCheckToken,
    String? firebaseAuthToken, {
    String? environment,
  }) {
    KangarooAppSdkInterface.instance.initializeSdk(
      appId: appId,
      environment: environment,
      firebaseAppCheckToken: firebaseAppCheckToken,
      firebaseAuthToken: firebaseAuthToken,
    );
  }

  static Future<String?> getPreferredLanguage() {
    return KangarooAppSdkInterface.instance.getPreferredLanguage();
  }

  static setPreferredLanguage(String preferredLanguage) {
    KangarooAppSdkInterface.instance.setPreferredLanguage(preferredLanguage);
  }

  static Future<String?> getSession() {
    return KangarooAppSdkInterface.instance.getSession();
  }

  static killSession() {
    KangarooAppSdkInterface.instance.killSession();
  }
}
